core/src/main/java/org/apache/commons/vfs2/provider/AbstractFileObject.java

```diff
@@ -564,24 +564,28 @@ private boolean deleteSelf() throws FileSystemException
      */
     private void detach() throws Exception
     {
-        synchronized (fs)
+        // no need to enter synchronized block if already detached.
```

Comment:
this is needed to avoid that the state can change while in the conditional block. It might be possible to use a DCL, but not sure if that helps much. VFS threading/concurrency is not really clear designed in some corners.